alpha/lib/model/entry.php

```diff
@@ -3117,15 +3117,27 @@ public function createDownloadAsset(BatchJob $parentJob = null, $flavorParamsId,
 	 */
 	public function getDownloadAssetUrl($flavorParamsId)
 	{
-		$flavorAsset = assetPeer::retrieveByEntryIdAndParams($this->getId(), $flavorParamsId);
-		if ($flavorAsset && $flavorAsset->getStatus() == flavorAsset::FLAVOR_ASSET_STATUS_READY)
-			return $flavorAsset->getDownloadUrl();
-			
-		if(assetPeer::retrieveOriginalByEntryId($this->getId()))
-			return null; // flavor asset should be created
-			
-		// entry file sync should be used (data or download)
-		return $this->getRawDownloadUrl();
+        	if($flavorParamsId == 0)
```

Comment:
===

Reply:
done

Reply:
Changed to === "0"
$flavorParamsId is string.

Reply:
Hi @bw-kaltura ,

I'm a bit confused as to why it is a string? Doxygen block declares it to be an `int`:
https://github.com/kaltura/server/blob/f04763dd877c7df2f2dd396b08d9e985c4a55dcf/alpha/lib/model/entry.php#L3115

The `flavor_asset.flavor_params_id` column is also defined as `int`:
https://github.com/kaltura/server/blob/Rigel-18.9.0/deployment/base/sql/01.kaltura_ce_tables.sql#L1073

And lastly:
https://github.com/kaltura/server/blob/Rigel-18.9.0/alpha/lib/model/om/BaseflavorParamsConversionProfile.php#L172